roles/compute/tasks/common.yml

```diff
@@ -0,0 +1,67 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+---
+- include_vars:
+    file: common.yml
+
+- name: Set layer's hostname
+  hostname:
+    name: compute
```

Comment:
For the application image, Alan M. was telling me to not set the hostname in the final image because we need the hostname to be set by DHCP. Otherwise we get say 2,000 compute nodes all with the same hostname. Maybe this is solved elsewhere with bootparams or something. Just pointing it out.

Reply:
This is just the hostname of the image, otherwise it's `install` when it boots. Showing `compute` just tells the developer or user that a compute image is booted, it's only seen by developers or when a deployment goes wrong.

This is fine.

What we need to actually be weary of [is this code](https://github.com/Cray-HPE/metal-provision/blob/main/roles/node-images-base/vars/metal.yml#L31). The compute role may want to override that value or we can move that single DHCP setting into the ncn-common role.